addons/purchase/tests/test_purchase_product_catalog.py

```diff
@@ -84,7 +84,7 @@ def test_catalog_price(self):
             headers={'Content-Type': 'application/json'},
         )
         self.assertEqual(resp.status_code, 200)
-        self.assertEqual(resp.json()['result'], other_product_price_converted)
+        self.assertEqual(resp.json()['result']['price'], other_product_price_converted)
```

Comment:
You should add more to this test to check your new `productUnitPrice` values are calculated correctly